autoload/go/cmd.vim

```diff
@@ -124,6 +135,35 @@ function! go#cmd#Test(compile, ...)
     endif
 endfunction
 
+function! go#cmd#TestFocused(...)
+    let test = search("func Test", "bcsW")
+
+    if test == 0
+        echo "vim-go: [test] no test found immediate to cursor"
+    else
+        let line = getline(test)
+        let name = split(split(line, " ")[1], "(")[0]
+        let flag = "-run '" . name . "$'"
+
+        let a1 = ""
+        if len(a:000)
+            let a1 = a:1
+
+            " add extra space
+            let flag = " " . flag 
+        endif
+
+        call go#cmd#Test(0, a1, flag)
+    end
+endfunction
+
+" default cache, in case TestLast is called before anything
+let s:go_cmd_test_args_cache = [0, ""]
+
+function! go#cmd#TestLast()
```

Comment:
I didn't like the caching functionality right now. May be it's because of the naming. It's good but it's somehow doesn't fit into the current code. In which cases are you using it?  (I'm curious about the usage)